as/cNFT/assembly/core.ts

```diff
@@ -37,14 +38,39 @@ export function nft_transfer(token_id: TokenId, bidder_id: AccountId): void {
     /* Deleting token from previous owner */
     persistent_tokens.remove(token.id, token.prev_owner_id)
 
-    
+
     // Immiting log event
     const transfer_log = new NftTransferLog()
-    
+
     transfer_log.old_owner_id = token.prev_owner_id
     transfer_log.new_owner_id = token.owner_id
     transfer_log.token_ids = [token.id]
 
     const log = new NftEventLogData<NftTransferLog>('nft_transfer', [transfer_log])
     logging.log(log)
+
+}
+
+
+@nearBindgen
+export function burn_design(token_id: TokenId): void {
+
+    /* Getting stored token from tokenId */
+    const token = persistent_tokens.get(token_id)
+
+    assert((context.sender == token.owner_id) && (context.sender == token.creator_id), "You must be the creator and the owner of the token to burn it")
+
+    /* Deleting token from its owner */
+    persistent_tokens.remove(token_id, token.owner_id);
```

Comment:
you are only removing it from account it token sent and not updating token to have `owner_id: ''` (empty string).